scripts/zkai_model_faithful_d128_block_boundary_preflight_gate.py

```
@@ -0,0 +1,746 @@
+            results.append({"name": name, "rejected": True, "error": str(err)})
+        else:
+            results.append({"name": name, "rejected": False, "error": None})
+    return results
+
+
+def resolve_output_path(path: pathlib.Path, base_dir: pathlib.Path) -> pathlib.Path:
+    candidate = pathlib.Path(os.path.abspath(path if path.is_absolute() else ROOT / path))
+    try:
+        candidate.relative_to(base_dir.resolve())
+    except ValueError as err:
+        raise ModelFaithfulD128BlockBoundaryPreflightError(f"output path must stay inside {base_dir}") from err
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)